plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/DeletionVectors.java

```diff
@@ -0,0 +1,270 @@
+import io.trino.spi.TrinoException;
+import org.roaringbitmap.RoaringBitmap;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.OptionalInt;
+import java.util.UUID;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static java.lang.Math.toIntExact;
+import static java.nio.ByteOrder.LITTLE_ENDIAN;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+// https://github.com/delta-io/delta/blob/master/PROTOCOL.md#deletion-vector-format
+public final class DeletionVectors
+{
+    private static final Logger log = Logger.get(DeletionVectors.class);
+    private static final int PORTABLE_ROARING_BITMAP_MAGIC_NUMBER = 1681511377;
+
+    private static final String UUID_MARKER = "u"; // relative path with random prefix on disk
+    private static final String PATH_MARKER = "p"; // absolute path on disk
+    private static final String INLINE_MARKER = "i"; // inline
+
+    private static final CharMatcher ALPHANUMERIC = CharMatcher.inRange('A', 'Z').or(CharMatcher.inRange('a', 'z')).or(CharMatcher.inRange('0', '9')).precomputed();
+
+    private DeletionVectors() {}
+
+    public static RoaringBitmap[] readDeletionVectors(
+            TrinoFileSystem fileSystem,
+            Location location,
+            String storageType,
+            String pathOrInlineDv,
+            OptionalInt offset,
+            int sizeInBytes,
+            long cardinality)
+            throws IOException
+    {
+        if (storageType.equals(UUID_MARKER)) {
+            TrinoInputFile inputFile = fileSystem.newInputFile(location.appendPath(toFileName(pathOrInlineDv)));
+            ByteBuffer buffer = readDeletionVector(inputFile, offset.orElseThrow(), sizeInBytes);
+            RoaringBitmap[] bitmaps = deserializeDeletionVectors(buffer);
+            if (bitmaps.length != cardinality) {
+                // Don't throw an exception because Databricks may report the wrong cardinality when there are many deleted rows
+                log.debug("The number of deleted rows expects %s but got %s", cardinality, bitmaps.length);
+            }
+            return bitmaps;
+        }
+        else if (storageType.equals(INLINE_MARKER) || storageType.equals(PATH_MARKER)) {
+            throw new TrinoException(NOT_SUPPORTED, "Unsupported storage type for deletion vector: " + storageType);
+        }
+        throw new IllegalArgumentException("Unexpected storage type: " + storageType);
+    }
+
+    public static String toFileName(String pathOrInlineDv)
+    {
+        int randomPrefixLength = pathOrInlineDv.length() - Base85Codec.ENCODED_UUID_LENGTH;
+        String randomPrefix = pathOrInlineDv.substring(0, randomPrefixLength);
+        checkArgument(ALPHANUMERIC.matchesAllOf(randomPrefix), "Random prefix must be alphanumeric: %s", randomPrefix);
```

Comment:
While testing, I haven't been able to create a scenario which makes use of a random prefix. If you have such a scenario, could you please add a table corresponding to it to `plugin/trino-delta-lake/src/test/resources/databricks/` with the corresponding test?